cocos/particle/renderer/particle-system-renderer-data.ts

```diff
@@ -151,12 +152,67 @@ export default class ParticleSystemRenderer {
         }
     }
 
+    /**
+     * @zh 粒子使用的cpu材质。
+     */
+    @type(Material)
+    @displayOrder(8)
+    @disallowAnimation
+    @visible(function (this: ParticleSystemRenderer): boolean { return !this._useGPU; })
+    public get cpuMaterial () {
+        return this._cpuMaterial;
+    }
+
+    public set cpuMaterial (val: Material | null) {
+        if (val === null) {
+            return;
+        } else {
+            const effectName = val.effectName;
+            if (effectName.indexOf('particle') === -1 || effectName.indexOf('particle-gpu') !== -1) {
```

Comment:
Why check "particle-gpu" ?

Reply:
do not use effect include gpu for cpu material

Reply:
> Why check "particle-gpu" ?

do not use effect include gpu for cpu material